src/wazuh_modules/wm_syscollector.c

```diff
@@ -151,6 +152,15 @@ void* wm_sys_main(wm_sys_t *sys) {
         syscollector_start_ptr = so_get_function_sym(syscollector_module, "syscollector_start");
         syscollector_stop_ptr = so_get_function_sym(syscollector_module, "syscollector_stop");
         syscollector_sync_message_ptr = so_get_function_sym(syscollector_module, "syscollector_sync_message");
+
+        void* rsync_module = NULL;
+        if(rsync_module = so_check_module_loaded("rsync"), rsync_module) {
+            void (*rsync_initialize_ptr)(full_log_fnc_t) = so_get_function_sym(rsync_module, "rsync_initialize_log_function");
```

Comment:
What about using typedef? I imagine the assignation would end like this: 
`rsync_initialize_ptr = so_get_function_sym(rsync_module, "rsync_initialize_log_function");`

Reply:
I've used a **typedef** now, please check.